tools/rule-notices.ts

```diff
@@ -0,0 +1,109 @@
+import plugin from '../src';
+
+enum MESSAGE_TYPE {
+  CONFIGS = 1,
+  DEPRECATED = 2,
+  FIXABLE = 3,
+  HAS_SUGGESTIONS = 4,
+}
+
+export const MESSAGES = {
+  [MESSAGE_TYPE.CONFIGS]:
+    '💼 This rule is enabled in the following [configs](https://github.com/jest-community/eslint-plugin-jest#shareable-configurations):',
+  [MESSAGE_TYPE.DEPRECATED]: '❌ This rule is deprecated.',
+  [MESSAGE_TYPE.FIXABLE]:
+    '🔧 This rule is automatically fixable using the `--fix` [option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix) on the command line.',
+  [MESSAGE_TYPE.HAS_SUGGESTIONS]:
+    '💡 This rule provides [suggestions](https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions) that can be applied manually.',
+};
+
+type RuleName = keyof typeof plugin.rules;
+type Rule = typeof plugin.rules[RuleName];
+
+/**
+ * Get config names that a given rule belongs to.
+ */
+function getConfigsForRule(ruleName: RuleName) {
+  const { configs } = plugin;
+  const configNames: Array<keyof typeof configs> = [];
+  let configName: keyof typeof configs;
+
+  for (configName in configs) {
+    const config = configs[configName];
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment -- we don't have a static type for rule names
+    // @ts-ignore
+    const value = config.rules[`jest/${ruleName}`];
+    const isEnabled = [2, 'error'].includes(value);
+
+    if (isEnabled) {
+      configNames.push(configName);
+    }
+  }
+
+  return configNames.sort();
+}
+
+/**
+ * Convert list of configs to formatted string list.
+ */
+function configNamesToList(configNames: string[]) {
+  return `\`${configNames.join('`, `')}\``;
+}
+
+/**
+ * Determine which notices should and should not be included at the top of a rule doc.
+ */
+export function getNoticesForRule(rule: Rule) {
+  const expectedNotices: MESSAGE_TYPE[] = [];
+  const unexpectedNotices: MESSAGE_TYPE[] = [];
+
+  if (rule.meta.deprecated) {
+    expectedNotices.push(MESSAGE_TYPE.DEPRECATED);
+    unexpectedNotices.push(MESSAGE_TYPE.CONFIGS);
+  } else {
+    unexpectedNotices.push(MESSAGE_TYPE.DEPRECATED);
+    expectedNotices.push(MESSAGE_TYPE.CONFIGS);
+  }
+  if (rule.meta.fixable) {
+    expectedNotices.push(MESSAGE_TYPE.FIXABLE);
+  } else {
+    unexpectedNotices.push(MESSAGE_TYPE.FIXABLE);
+  }
+  if (rule.meta.hasSuggestions) {
+    expectedNotices.push(MESSAGE_TYPE.HAS_SUGGESTIONS);
+  } else {
+    unexpectedNotices.push(MESSAGE_TYPE.HAS_SUGGESTIONS);
+  }
+
+  return {
+    expectedNotices,
+    unexpectedNotices,
+  };
+}
+
+/**
+ * Get the lines for the notice section at the top of a rule doc.
+ */
+export function getRuleNoticeLines(ruleName: RuleName) {
+  const lines: string[] = [];
+
+  const { expectedNotices } = getNoticesForRule(plugin.rules[ruleName]);
+
+  expectedNotices.forEach(expectedNotice => {
```

Comment:
nit: I'd prefer using `for ... of`

Reply:
Fixed. I recommend we enable eslint-plugin-unicorn recommended rules including [unicorn/no-array-for-each](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-for-each.md) to catch stuff like this.

Reply:
I'm not committed to it enough yet to bring in a whole new plugin especially just for one rule (+ I don't think its worth the energy bikeshedding over all the rules we could have, given how stable this plugin is) - its more something I've been evaluating for the last couple of months to see how I feel about it :)

Reply:
Well this is just one example. Of course I think there are many valuable rules in eslint-plugin-unicorn. Something to consider separately.